### PR TITLE
Make confusion_matrix and roc generic

### PR DESCRIPTION
**Pull request by Jeffwan**
Add s3 and minio storage options in confusion_matrix and roc components.
Frontend s3 support has been added in this PR https://github.com/kubeflow/pipelines/pull/1278

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1285)

### CODE REVIEW

File: components/local/confusion_matrix/src/confusion_matrix.py [hunk 1]
```diff
@@ -40,7 +41,9 @@ def main(argv=None):
                            'If not set, the input must include a "target" column.')
   args = parser.parse_args()
 
-  on_cloud = args.output.startswith('gs://')
+  storage_service_dict = {'minio': 'minio', 'gs': 'gcs', 's3': 's3', '': 'gcs'}
```
Comment (Jeffwan) [inline components/local/confusion_matrix/src/confusion_matrix.py]: local mode can not be supported in frontend. We leave `gcs` as default here.

**Comment by Ark-kun** [comment]
/lgtm
But can this be solved in a more generic way? Can the front-end deduce the storage information?

**Comment by animeshsingh** [comment]
Can minio client not frontend any s3 type storage? Do we need them separately?

**Comment by Jeffwan** [comment]
> Can minio client not frontend any s3 type storage? Do we need them separately?

Yes. Frontend will check protocol for file in `/mlpipeline-ui-metadata.json`. It makes calls to minio or S3 correspondingly.

**Comment by Jeffwan** [comment]
> /lgtm
> But can this be solved in a more generic way? Can the front-end deduce the storage information?

I double checks code. Actually, `storage` is not being used for roc and confusion matrix.  The only reference for this filed is here for MarkdownViewer.

https://github.com/kubeflow/pipelines/blob/892e3082b748b5029bf218a7dfee73af6abf596b/frontend/src/lib/OutputArtifactLoader.ts#L202

I think it's safe to remove this field on two files. Frontend will get storage service from file path. What do you think?

**Comment by Ark-kun** [comment]
> I double checks code. Actually, `storage` is not being used for roc and confusion matrix. The only reference for this filed is here for MarkdownViewer.
> 
> https://github.com/kubeflow/pipelines/blob/892e3082b748b5029bf218a7dfee73af6abf596b/frontend/src/lib/OutputArtifactLoader.ts#L202
> 
> Verified it's safe to remove this field on two files. Frontend will get storage service from file path. What do you think?

/cc @rileyjbauer @gaoning777

**Comment by rileyjbauer** [comment]
> I double check code. Actually, `storage` is not being used for roc and confusion matrix. The only reference for this filed is here for MarkdownViewer.
> 
> https://github.com/kubeflow/pipelines/blob/892e3082b748b5029bf218a7dfee73af6abf596b/frontend/src/lib/OutputArtifactLoader.ts#L202
> 
> Verified it's safe to remove this field on two files. Frontend will get storage service from file path. What do you think?

Can you clarify this? What two files are you referring to?

Like you mentioned, in `OutputArtifactLoader.ts` the confusion matrix and ROC code doesn't currently use the output metadata `storage` field at all, it just uses the `source`. Does that mean additional frontend changes will be needed after this PR to make use of the now updated `storage` fields?

**Comment by Jeffwan** [comment]
> Can you clarify this? What two files are you referring to?

two files I mean are 
https://github.com/kubeflow/pipelines/blob/master/components/local/confusion_matrix/src/confusion_matrix.py and
https://github.com/kubeflow/pipelines/blob/master/components/local/roc/src/roc.py

These two files write unneeded `storage` into manifest.

> Like you mentioned, in `OutputArtifactLoader.ts` the confusion matrix and ROC code doesn't currently use the output metadata `storage` field at all, it just uses the `source`. Does that mean additional frontend changes will be needed after this PR to make use of the now updated `storage` fields?

At the beginning, I think frontend may use `storage` and then I added different options. Now, after your confirmation, I think we can just remove `storage` field. Different storage should be supported out-of-box. Frontend doesn't need to make any changes. Otherwise, I think users may have confusion, they are only working with GCS?

**Comment by rileyjbauer** [comment]
> Now, after your confirmation, I think we can just remove `storage` field. Different storage should be supported out-of-box. Frontend doesn't need to make any changes. Otherwise, I think users may have confusion, they are only working with GCS? (I was confused and then file this PR)
I see. So you'll change this PR to just remove `storage` in these python files to prevent confusion? Sounds good to me.

File: components/local/confusion_matrix/src/confusion_matrix.py [hunk 2]
```diff
@@ -40,7 +41,8 @@ def main(argv=None):
                            'If not set, the input must include a "target" column.')
   args = parser.parse_args()
 
-  on_cloud = args.output.startswith('gs://')
+  storage_service_scheme = urlparse.urlparse(args.output).scheme
```
Comment (Jeffwan) [inline components/local/confusion_matrix/src/confusion_matrix.py]: a few cases are 

- minio://dir/file_path 
- gs://dir/file_path  
- s3://dir/file_path
- local path "/dir/file_path"

scheme for local path is empty, all rest we consider they are `on_cloud`

**Comment by Jeffwan** [comment]
> > Now, after your confirmation, I think we can just remove `storage` field. Different storage should be supported out-of-box. Frontend doesn't need to make any changes. Otherwise, I think users may have confusion, they are only working with GCS? (I was confused and then file this PR)
> 
> I see. So you'll change this PR to just remove `storage` in these python files to prevent confusion? Sounds good to me.

Cool. Make the change.

**Pull request merged**